samples/src/main/java/com/microsoft/applicationinsights/sample/plugins/GitBuildInfoContextInitializer.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * AppInsights-Java
+ * Copyright (c) Microsoft Corporation
+ * All rights reserved.
+ *
+ * MIT License
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this
+ * software and associated documentation files (the ""Software""), to deal in the Software
+ * without restriction, including without limitation the rights to use, copy, modify, merge,
+ * publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+ * persons to whom the Software is furnished to do so, subject to the following conditions:
+ * The above copyright notice and this permission notice shall be included in all copies or
+ * substantial portions of the Software.
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR
+ * PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE
+ * FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+ * OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+
+package com.microsoft.applicationinsights.sample.plugins;
+
+import java.io.InputStream;
+import java.util.Properties;
+
+import com.microsoft.applicationinsights.extensibility.ContextInitializer;
+import com.microsoft.applicationinsights.internal.logger.InternalLogger;
+import com.microsoft.applicationinsights.telemetry.TelemetryContext;
+
+/**
+ * An initializer to fetch Git data from a properties file
+ * Note: the file with its full package name should appear in the 'ApplicationInsights.xml':
+ *
+ * <ContextInitializers>
+ *   <Add type="com.microsoft.applicationinsights.sample.plugins.GitBuildInfoContextInitializer" />
+ * </ContextInitializers>
+ */
+public class GitBuildInfoContextInitializer implements ContextInitializer {
+    private final static String BUILD_INFO_FILE_NAME = "source-origin.properties";
+
+    final static String GIT_BRANCH_NAME = "git.branch";
+    final static String GIT_COMMIT_NAME = "git.commit";
+    final static String GIT_REPO_NAME = "git.repo";
+    final static String GIT_URL_NAME = "git.url";
```

Comment:
As with this implementation the URL represents the URL for the commit then I'd change it to be:

    final static String GIT_COMMIT_URL_NAME = "git.commit.url";